plugin.video.mlbtv/service.py

```diff
@@ -0,0 +1,123 @@
+import threading
+
+import xbmc
+import requests
+
+try:
+    # Python3
+    from http.server import BaseHTTPRequestHandler, HTTPServer
+    from socketserver import ThreadingMixIn
+    from urllib.parse import urljoin
+except:
+    # Python2
+    from BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
+    from SocketServer import ThreadingMixIn
+    from urlparse import urljoin
+
+HOST = '127.0.0.1'
+PORT = 43670
+PROXY_URL = 'http://' + HOST + ':' + str(PORT) + '/'
+STREAM_EXTENSION = '.m3u8'
+URI_START_DELIMETER = 'URI="'
+URI_END_DELIMETER = '"'
+KEY_TEXT = '-KEY:METHOD=AES-128'
+ENDLIST_TEXT = '#EXT-X-ENDLIST'
+REMOVE_IN_HEADERS = ['upgrade', 'host', 'accept-encoding', 'pad']
+REMOVE_OUT_HEADERS = ['date', 'server', 'transfer-encoding', 'keep-alive', 'connection', 'content-length', 'content-md5', 'access-control-allow-credentials', 'content-encoding']
+
+class RequestHandler(BaseHTTPRequestHandler):
```

Comment:
Why do you need this HTTP server?

Reply:
I believe this so the stream can be altered to have random lengths. Description is below and link to the PR with even more info is at the bottom.

_No more timeline spoilers! When spoilers are hidden for an archive stream, the stream gets padded with a random amount of extra time on the end, so you don't know exactly how long the game has left to go. This is accomplished by running the streams through a local streaming proxy service.py based on [this example](https://github.com/matthuisman/proxy.plugin.example)_

https://github.com/eracknaphobia/plugin.video.mlbtv/pull/44